type-check/throwIf.test.ts

```diff
@@ -0,0 +1,81 @@
+import { randomInt } from "crypto";
+
+import { isNil, pipe, throwIf } from "../src";
+import * as Test from "../src/types/Test";
+
+const { checks, check } = Test;
+
+interface User {
+  name?: string | null;
+  age: number;
+}
+
+const createUser = (): User => ({ name: null, age: 23 });
+
+const res1 = createUser();
+const res2 = pipe(
+  createUser(),
+
+  (user) => user.name,
+
+  throwIf<string | null | undefined, null | undefined>(isNil, (name) =>
+    Error(`User name is ${typeof name}`),
+  ),
```

Comment:
One of the main design goals of the FxTS library is to allow writing code in a form that closely resembles pure JavaScript, without directly using type parameters whenever possible. I think this use-case can also easily align with that design goal.

```suggestion
  throwIf(isNil, (name) => Error(`User name is ${typeof name}`)),
```

Reply:
위에 언급했듯이 명시하지 않으면 타입가드 형식의 오버로딩이 적용되지 않습니다. 그래서 타입가드 형식만 허용하도록 수정할 예정입니다